### ISSUE TITLE
Any way to close ALL active subscriptions?

### ISSUE THREAD

**Issue by slumtrimpet** [gqless/gqless#208]
Our GQLess implementation involves numerous run-time updates to subscription parameters (to facilitate user-requested sort/filter in a table view).

That becomes problematic as subscriptions seem to 'accumulate' as our query parameters are updated.  This is in spite of us calling the 'unsubscribe' methods made available on the subscription events themselves.  Something is still remaining 'subscribed.  

After a user sorts or filters their report a few times, they start receiving many instances of each subscription event instead of just the current one they should.

We've found that executing ```subscriptionsClient.close()``` before modifying the active subscription seems to be the resolution as it results in one single subscription open at all times.  The problem is that the auto-generated GQLess schema file does not seem to export any access to the actual SubscriptionsClient object.  

Long story short... this is either an implementation issue on us, an unintended use case in GQLess, or all is good and we just need the auto-generated schema file to contain the following:

```
export const subscriptionsClient =
  typeof window !== 'undefined'
    ? createSubscriptionsClient({
```

Instead of the current:

```
const subscriptionsClient =
  typeof window !== 'undefined'
    ? createSubscriptionsClient({
```

**Comment by PabloSzx** [comment]
Using subscriptionsClient.close() is completely possible, it's there for a reason, but for simplicity reasons it's not exported.

> or all is good and we just need the auto-generated schema file to contain the following

The subscriptions client is not generated in the schema file, it is generated in the client file, which is intended to be modified, there is nothing wrong in that.

> After a user sorts or filters their report a few times, they start receiving many instances of each subscription event instead of just the current one they should

The current implementation of the unsubscribe functions of "resolved" only take in account the operations that are made in a single pass of scalars selections, if you add selections outside of it, those functions won't be able to it's thing with the new generated subscriptions, and if you are using "useSubscription" that issue shouldn't exist.

If you feel that there is a bug on the implementation, a reproduction would be very appreciated 👍

**Comment by slumtrimpet** [comment]
> Using subscriptionsClient.close() is completely possible, it's there for a reason, but for simplicity reasons it's not exported.
> 
> > or all is good and we just need the auto-generated schema file to contain the following
> 
> The subscriptions client is not generated in the schema file, it is generated in the client file, which is intended to be modified, there is nothing wrong in that.
> 
> > After a user sorts or filters their report a few times, they start receiving many instances of each subscription event instead of just the current one they should
> 
> The current implementation of the unsubscribe functions of "resolved" only take in account the operations that are made in a single pass of scalars selections, if you add selections outside of it, those functions won't be able to it's thing with the new generated subscriptions, and using "useSubscription" with React, that issue shouldn't exist.
> 
> If you feel that there is a bug on the implementation, a reproduction would be very appreciated 👍

This makes sense to me.  I, for some reason, had ignored the comment in the auto-generated GQless Schema file:

```
/**
 * GQLESS: You can safely modify this file and Query Fetcher based on your needs
 */
```

Now that I realize I can take ownership of that file, I'm proceeding without any issues.  Thanks for your support!
